projects/components/src/color-picker/color-picker.component.ts

```diff
@@ -58,9 +58,12 @@ export class ColorPickerComponent implements ControlValueAccessor {
   private propagateControlValueChange?: (value: string | undefined) => void;
   private propagateControlValueChangeOnTouch?: (value: string | undefined) => void;
 
+  public ngOnInit(): void {
```

Comment:
OnChanges please - selected can be changed

Reply:
Yes, but also can be changed after clicking on a new color and I don't. want this logic.

Reply:
Yes, it can be changed by clicking on a color, which calls `selectColor` (we know in this case it's already part of the palette), via input (the onChanges case) or via the programmatic writeValue api. In the first one we'd emit (already correct) and in the last two we wouldn't, but we'd need to make sure it's part of the palette. All that I believe is correct here. The issue is that rather than looking for changes whenever component input changes via OnChanges, we're only looking once via OnInit.

Reply:
clear. done